crates/aptos/src/account/key_rotation.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeMap;
+
+use crate::common::{
+    types::{
+        CliCommand, CliConfig, CliError, CliTypedResult, ConfigSearchMode, EncodingOptions,
+        ExtractPublicKey, ProfileConfig, ProfileOptions, PromptOptions, PublicKeyInputOptions,
+        RestOptions, TransactionOptions, TransactionSummary,
+    },
+    utils::{prompt_yes_with_override, read_line},
+};
+use aptos_crypto::{
+    ed25519::{Ed25519PrivateKey, Ed25519PublicKey},
+    PrivateKey, SigningKey,
+};
+use aptos_rest_client::Client;
+use aptos_types::{
+    account_address::AccountAddress, account_config::CORE_CODE_ADDRESS,
+    transaction::authenticator::AuthenticationKey,
+};
+use async_trait::async_trait;
+use cached_packages::aptos_stdlib;
+use clap::Parser;
+use serde::{Deserialize, Serialize};
+
+// This struct includes TypeInfo (account_address, module_name, and struct_name)
+// and RotationProofChallenge-specific information (sequence_number, originator, current_auth_key, and new_public_key)
+// Since the struct RotationProofChallenge is defined in "0x1::account::RotationProofChallenge",
+// we will be passing in "0x1" to `account_address`, "account" to `module_name`, and "RotationProofChallenge" to `struct_name`
+// Originator refers to the user's address
+#[derive(Serialize, Deserialize)]
+pub struct RotationProofChallenge {
+    // Should be `CORE_CODE_ADDRESS`
+    pub account_address: AccountAddress,
+    // Should be `account`
+    pub module_name: String,
+    // Should be `RotationProofChallenge`
+    pub struct_name: String,
+    pub sequence_number: u64,
+    pub originator: AccountAddress,
+    pub current_auth_key: AccountAddress,
+    pub new_public_key: Vec<u8>,
+}
+
+/// Command to rotate the account auth key
+///
+#[derive(Debug, Parser)]
+pub struct RotateKey {
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+
+    #[clap(flatten)]
+    pub(crate) prompt_options: PromptOptions,
+
+    /// Private key encoded in a type as shown in `encoding`
+    #[clap(long)]
+    pub(crate) new_private_key: String,
```

Comment:
Okay, so the private key has to be typed in via the CLI input, if that's the case this will show up in a shell history, which means it ends up being much more insecure.

Maybe we provide the file input as well?